soc/arm/nordic_nrf/nrf51/power.c

```diff
@@ -1,11 +1,95 @@
 /*
- * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2017 Intel Corporation.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 #include <zephyr.h>
+#include <soc_power.h>
 #include <nrf_power.h>
 
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Invoke Low Power/System Off specific Tasks */
+void sys_set_power_state(enum power_states state)
+{
+	switch (state) {
+#ifdef CONFIG_SYS_POWER_DEEP_SLEEP
+#ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED
+	case SYS_POWER_STATE_DEEP_SLEEP:
+		nrf_power_system_off();
+		break;
+#endif
+#endif
+	default:
+		LOG_ERR("Unsupported power state %u", state);
+		break;
+	}
+}
+
+/* Handle SOC specific activity after Low Power Mode Exit */
+void sys_power_state_post_ops(enum power_states state)
+{
+	switch (state) {
+#ifdef CONFIG_SYS_POWER_DEEP_SLEEP
+# ifdef CONFIG_SYS_POWER_STATE_DEEP_SLEEP_SUPPORTED
+	case SYS_POWER_STATE_DEEP_SLEEP:
+		/* Nothing to do. */
+		break;
+# endif
+#endif
+	default:
+		LOG_ERR("Unsupported power state %u", state);
+		break;
+	}
+
+	/*
+	 * System is now in active mode. Reenable interrupts which were disabled
+	 * when OS started idling code.
+	 */
+	__enable_irq();
```

Comment:
Just wondering: Where exactly are the interrupts disabled?

Reply:
OK, I suppose when you need to implement also for 9160, this will be even more demanding.